mipengine/udfgen/udfgen_DTOs.py

```diff
@@ -0,0 +1,150 @@
+from string import Template
+from typing import List
+from typing import Optional
+
+from mipengine.node_tasks_DTOs import TableInfo
+
+
+class UDFGenResult:
+    pass
+
+
+class TableUDFGenResult(UDFGenResult):
+    tablename_placeholder: str
+    drop_query: Template
+    create_query: Template
+
+    def __init__(
+        self,
+        tablename_placeholder,
+        drop_query,
+        create_query,
+    ):
+        self.tablename_placeholder = tablename_placeholder
+        self.drop_query = drop_query
+        self.create_query = create_query
+
+    def __eq__(self, other):
+        if self.tablename_placeholder != other.tablename_placeholder:
+            return False
+        if self.drop_query.template != other.drop_query.template:
+            return False
+        if self.create_query.template != other.create_query.template:
+            return False
+        return True
+
+    def __repr__(self):
+        return (
+            f"TableUDFGenResult("
+            f"tablename_placeholder='{self.tablename_placeholder}', "
+            f"drop_query='{self.drop_query.template}', "
+            f"create_query='{self.create_query.template}'"
+            f")"
+        )
+
+
+class SMPCUDFGenResult(UDFGenResult):
+    template: TableUDFGenResult
+    add_op_values: TableUDFGenResult
+    min_op_values: TableUDFGenResult
+    max_op_values: TableUDFGenResult
+    union_op_values: TableUDFGenResult
+
+    def __init__(
+        self,
+        template,
+        add_op_values=None,
+        min_op_values=None,
+        max_op_values=None,
+        union_op_values=None,
+    ):
+        self.template = template
+        self.add_op_values = add_op_values
+        self.min_op_values = min_op_values
+        self.max_op_values = max_op_values
+        self.union_op_values = union_op_values
+
```

Comment:
same here, declaring class/static vars `template,add_op_values,min_op_values,max_op_values, union_op_values` and then also declaring instance variables for the same attributes

Reply:
https://www.python.org/dev/peps/pep-0526/#class-and-instance-variable-annotations

Type annotation does not make a variable a class variable.

Reply:
Fixed.